### ISSUE TITLE
Running app locally with nodejs:12 fails with error

### ISSUE THREAD

**Issue by honstar** [adobe/aio-cli-plugin-app#287]
**Describe the bug**
When running a `nodejs:12` runtime app locally via `aio app run --local`, calling the action via cUrl for example fails with a 400 bad request response. The corresponding activation status suggests an issue outside of the action code itself:
```
"response": {
    "result": {
      "error": "The action did not initialize and exited unexpectedly."
    },
    "status": "action developer error",
    "success": false
  },
```

**To Reproduce**
1. Init a boilerplate app via `aio app init`
2. Make sure that the action is set to use `nodejs:12` in `manifest.yml`
3. Run the app locally via `aio app run --local`
4. Request the action via cUrl
5. Observe the response and activation status

**Expected behavior**
The action should run successfully.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS Catalina
 - Browser: Chrome
 - Version: 85.0

**Comment by meryllblanchet** [comment]
Hi @honstar and thanks for submitting the issue.
I confirm that I can reproduce it, with both `openwhisk/action-nodejs-v12:latest` and `adobeapiplatform/adobe-action-nodejs-v12:3.0.22` pulled on Docker Desktop 2.3.0.3.

Just for the sake of completeness, could I ask you to share with us the output of `aio info`?

**Comment by honstar** [comment]
```
$ aio info

  System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 159.44 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.18.3 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.14.6 - /usr/local/bin/npm
  Virtualization:
    Docker: 19.03.12 - /usr/local/bin/docker
  npmGlobalPackages:
    @adobe/aio-cli: 3.7.0
  CLI plugins:
    @adobe/aio-cli 3.7.0 (core)
    @adobe/aio-cli-plugin-info 1.1.0 (user)
    @adobe/aio-cli-plugin-console 3.2.0 (user)
    @adobe/aio-cli-plugin-app 3.5.0 (user)
    @adobe/aio-cli-plugin-auth 2.4.0 (user)
    @adobe/aio-cli-plugin-runtime 1.10.0 (user)
    @oclif/plugin-help 2.2.3 (core)
    @oclif/plugin-plugins 1.9.0 (core)
    @oclif/plugin-autocomplete 0.2.0 (core)
    @oclif/plugin-warn-if-update-available 1.7.0 (core)
    @oclif/plugin-not-found 1.2.4 (core)
    @adobe/aio-cli-plugin-config 2.3.0 (core)
    @adobe/aio-cli-plugin-certificate 0.2.0 (core)
    @adobe/aio-cli-plugin-events 1.1.0 (core)
```

**Comment by moritzraho** [comment]
FYI, as a debugging starting point this PR that addressed the issue in the past could be helpful, https://github.com/adobe/aio-lib-web/pull/122

**Comment by shazron** [comment]
Verified the bug.

I tried updating the Docker images:
```
docker pull openwhisk/action-nodejs-v12:latest
docker pull adobeapiplatform/adobe-action-nodejs-v12:3.0.22
```
but doesn't fix the result.

I also verified the `runtimes.json` file has the `nodejs:12` kind setup properly. I'll have to debug the java openwhisk.jar logging output itself.

**Comment by shazron** [comment]
The log output doesn't show any errors, the containers for nodejs:12 are launched successfully. A simple action also runs, so it must be our boilerplate code.

**Comment by shazron** [comment]
I believe I've isolated the problem. I tested with the `generic` action template, and commenting out this line (and all the log lines that use it), makes the action successfully run:
```
const { Core } = require('@adobe/aio-sdk')
```
Investigating further.

**Comment by shazron** [comment]
Loading the aio-sdk locally in node-12 is fine, so it must be a webpack issue.

**Comment by shazron** [comment]
I added `@adobe/aio-lib-files` to the app package.json, deployed and tried to load it in the generic template. This causes the 400 error. Removing the require made it work again.

```
require('@adobe/aio-lib-files')
```

Looking at warnings in @adobe/aio-lib-files, there is a warning:
```
npm WARN deprecated crypto@1.0.1: This package is no longer supported. It's now a built-in Node module. If you've depended on crypto, you should switch to the one that's built-in.
```
... not sure if it would affect things.

**Comment by shazron** [comment]
The activation data does not give any clues:
```
ok: got activation 469ba1712af14bab9ba1712af1cbaba4
{
    "namespace": "guest",
    "name": "foo",
    "version": "0.0.3",
    "subject": "guest",
    "activationId": "469ba1712af14bab9ba1712af1cbaba4",
    "start": 1605674048987,
    "end": 1605674049010,
    "duration": 23,
    "statusCode": 0,
    "response": {
        "status": "action developer error",
        "statusCode": 0,
        "success": false,
        "result": {
            "error": "The action did not initialize and exited unexpectedly."
        }
    },
    "logs": [],
    "annotations": [
        {
            "key": "path",
            "value": "guest/realisticazureboar-0.0.1/foo"
        },
        {
            "key": "waitTime",
            "value": 954
        },
        {
            "key": "kind",
            "value": "nodejs:12"
        },
        {
            "key": "timeout",
            "value": false
        },
        {
            "key": "limits",
            "value": {
                "concurrency": 1,
                "logs": 10,
                "memory": 256,
                "timeout": 60000
            }
        },
        {
            "key": "initTime",
            "value": 23
        }
    ],
    "publish": false
}
```

**Comment by shazron** [comment]
Trying 4 different things on local:
1. deploy to remote server (✅ it works on prod)
2. load a different  module that has node-fetch as a dependency
3. run the same tests on Linux
4. run the same tests on Windows

**Comment by shazron** [comment]
Ignore the results above, turns out to be a red herring 😅 
The 400 error is a result of the Docker container failing to be invoked:
```
[2020-11-19T18:39:14.722Z] [INFO] [#tid_XfMBpnM5ZIPH9bFf67PhHztYe5mMIG0k] [DockerContainer] initialization result: ConnectionError(java.net.SocketException: Broken pipe) [marker:invoker_activationInit_finish:15815:67]
[2020-11-19T18:39:14.722Z] [INFO] [#tid_XfMBpnM5ZIPH9bFf67PhHztYe5mMIG0k] [ContainerProxy]  [marker:invoker_collectLogs_start:15817]
[2020-11-19T18:39:14.722Z] [INFO] [#tid_XfMBpnM5ZIPH9bFf67PhHztYe5mMIG0k] [ExtendedDockerClient] running /usr/local/bin/docker logs c81c5fb380f2e0340ad4922ac06a37305020dd922c6e536fc46325dd2f493400 --since 1605782342 --until 1605782343 --timestamps (timeout: 2 seconds) [marker:invoker_docker.logs_start:15818]
```

This error is intermittent, and I'm not sure why the Docker invocation is failing. After a few tries sometimes, it starts working again (the same 400 error is shown if Docker is not running, as an example)

**Comment by hardikthakkar2905** [comment]
> ```
> $ aio info
> 
>   System:
>     OS: macOS 10.15.6
>     CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
>     Memory: 159.44 MB / 16.00 GB
>     Shell: 3.2.57 - /bin/bash
>   Binaries:
>     Node: 12.18.3 - /usr/local/bin/node
>     Yarn: Not Found
>     npm: 6.14.6 - /usr/local/bin/npm
>   Virtualization:
>     Docker: 19.03.12 - /usr/local/bin/docker
>   npmGlobalPackages:
>     @adobe/aio-cli: 3.7.0
>   CLI plugins:
>     @adobe/aio-cli 3.7.0 (core)
>     @adobe/aio-cli-plugin-info 1.1.0 (user)
>     @adobe/aio-cli-plugin-console 3.2.0 (user)
>     @adobe/aio-cli-plugin-app 3.5.0 (user)
>     @adobe/aio-cli-plugin-auth 2.4.0 (user)
>     @adobe/aio-cli-plugin-runtime 1.10.0 (user)
>     @oclif/plugin-help 2.2.3 (core)
>     @oclif/plugin-plugins 1.9.0 (core)
>     @oclif/plugin-autocomplete 0.2.0 (core)
>     @oclif/plugin-warn-if-update-available 1.7.0 (core)
>     @oclif/plugin-not-found 1.2.4 (core)
>     @adobe/aio-cli-plugin-config 2.3.0 (core)
>     @adobe/aio-cli-plugin-certificate 0.2.0 (core)
>     @adobe/aio-cli-plugin-events 1.1.0 (core)
> ```

@honstar this helped. Running this command highlighted that I was not using correct node version. Using NVM to use correct version and then doing aio CLI install again after switching node version fixed it for me.
